### PR TITLE
Cleaned up edit text file

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Tested on windows and works fine

### CODE REVIEW

File: core/edit_text_file.py [hunk 1]
```diff
@@ -16,38 +15,29 @@
...
-    "search engines": os.path.join(
-        REPO_DIR, "core", "websites_and_search_engines", "search_engine.talon-list"
-    ),
-    "unix utilities": os.path.join(
-        REPO_DIR, "tags", "terminal", "unix_utility.talon-list"
-    ),
-    "websites": os.path.join(
-        REPO_DIR, "core", "websites_and_search_engines", "website.talon-list"
-    ),
+    "additional words": "core/vocabulary/vocabulary.talon-list",
```
Comment (knausj85) [inline core/edit_text_file.py:18]: I wonder if we should just move this stuff to a talon list while we're in here?

Comment (AndreasArvidsson) [reply core/edit_text_file.py]: done!

**Comment by phillco** [comment]
I get this on startup. Is it spurious?

```talon.scripting.types.ActionImplError: Function 'user.edit_text_file' incompatible with prototype 'user.edit_text_file'
  Signature: (file: str)
  Expected:  (path: str)
  - Parameter 'file' name incorrect, expected 'path'```

File: core/edit_text_file/edit_text_file.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import os
+import subprocess
+from pathlib import Path
+
+from talon import Context, Module, app
+
+# Path to community root directory
+REPO_DIR = os.path.dirname(os.path.dirname(__file__))
```
Comment (nriley) [inline core/edit_text_file/edit_text_file.py:8]: Please change this to use pathlib.

**Review by nriley — CHANGES_REQUESTED** [review]

**Comment by phillco** [comment]
Hm, I noticed an issue when testing this with a separate instance. user pwd should be `/Users/phillco/.talon-community/.talon/user/community` but it tries `/Users/phillco/.talon-community/.talon/core/keys/letter.talon-list`. I don't know if it's this change though, hasn't used this code before.

```The file /Users/phillco/.talon-community/.talon/core/keys/letter.talon-list does not exist.
2024-11-30 11:05:14.434 ERROR     8:        talon/scripting/talon_script.py:721|
    7:        talon/scripting/talon_script.py:314|
    6:             talon/scripting/actions.py:88 |
    5: user/community2/core/edit_text_file.py:76 | open_with_subprocess(
    4:                                             ^
    3: user/community2/core/edit_text_file.py:98 | return subprocess.run(args, timeout=0...
    2:                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    1:           lib/python3.11/subprocess.py:571|
subprocess.CalledProcessError: Command '['/usr/bin/open', '-t', PosixPath('/Users/phillco/.talon-community/.talon/core/keys/letter.talon-list')]' returned non-zero exit status 1.
```

File: core/edit_text_file/edit_text_file.py (hunk 2 above)
Comment (AndreasArvidsson) [reply core/edit_text_file/edit_text_file.py]: done

**Comment by nriley** [comment]
Holding off on merging this as @phillco is having an issue where this is triggering the type checker in a confusing way.

**Review by phillco — APPROVED** [review]
Works for me now.

**Review by nriley — APPROVED** [review]

**Pull request merged**